sound/soc/sdca/sdca_functions.c

```diff
@@ -30,5 +30,85 @@ int sdca_alloc(void **context,
 }
 EXPORT_SYMBOL_NS(sdca_alloc, SND_SOC_SDCA);
 
-MODULE_LICENSE("Dual BSD/GPL");
+static int find_sdca_function(struct acpi_device *adev, void *data)
+{
+	struct fwnode_handle *function_node = acpi_fwnode_handle(adev);
+	struct sdca_data *sdca = data;
+	struct device *parent = sdca->parent;
+	struct fwnode_handle *control5; /* used to identify topology type */
+	bool valid_function_found = false;
+	u32 function_topology = 0;
+	acpi_status status;
+	u64 addr;
+
+	status = acpi_evaluate_integer(adev->handle,
+				       METHOD_NAME__ADR, NULL, &addr);
```

Comment:
Should probably acpi_get_local_address

Reply:
Erm, no. that function is wrong. _ADR is a 64-bit value, but the code assumes it's 32-bits.

I vaguely remember we had to fix some of the ACPI subsystem to make _ADR a true 64 bit value, this function has not been updated apparently.

Actually now I do remember, and it's even in git:

commit ca6f998cf9a259fe6019d44768064e0cfe8a925b
Author: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date:   Wed May 1 07:53:22 2019 -0500

    ACPI: bus: change _ADR representation to 64 bits

So this code would need to be updated to u64.

````c
int acpi_get_local_address(acpi_handle handle, u32 *addr)
{
	unsigned long long adr;
	acpi_status status;

	status = acpi_evaluate_integer(handle, METHOD_NAME__ADR, NULL, &adr);
	if (ACPI_FAILURE(status))
		return -ENODATA;

	*addr = (u32)adr;
	return 0;
}
````

Gah.